src/main/java/appeng/block/AEBaseTileBlock.java

```diff
@@ -271,87 +271,93 @@ public void onBlockPlacedBy( final World w, final BlockPos pos, final IBlockStat
 	@Override
 	public boolean onBlockActivated( final World w, final BlockPos pos, final IBlockState state, final EntityPlayer player, final EnumHand hand, final @Nullable ItemStack heldItem, final EnumFacing side, final float hitX, final float hitY, final float hitZ )
 	{
-		if( player != null )
+		if( player != null && heldItem != null )
 		{
-			if( heldItem != null )
+			if( Platform.isWrench( player, heldItem, pos ) && player.isSneaking() )
 			{
-				if( Platform.isWrench( player, heldItem, pos ) && player.isSneaking() )
+				final IBlockState blockState = w.getBlockState( pos );
+				final Block block = blockState.getBlock();
+
+				if( block == null )
 				{
-					final IBlockState ids = w.getBlockState( pos );
-					final Block id = ids.getBlock();
-					if( id != null )
-					{
-						final AEBaseTile tile = this.getTileEntity( w, pos );
-						final ItemStack[] drops = Platform.getBlockDrops( w, pos );
+					return false;
+				}
 
-						if( tile == null )
-						{
-							return false;
-						}
+				final AEBaseTile tile = this.getTileEntity( w, pos );
 
-						if( tile instanceof TileCableBus || tile instanceof TileSkyChest )
-						{
-							return false;
-						}
+				if( tile == null )
+				{
+					return false;
+				}
 
-						final ItemStack op = new ItemStack( this );
-						for( final ItemStack ol : drops )
-						{
-							if( Platform.isSameItemType( ol, op ) )
-							{
-								final NBTTagCompound tag = tile.downloadSettings( SettingsFrom.DISMANTLE_ITEM );
-								if( tag != null )
-								{
-									ol.setTagCompound( tag );
-								}
-							}
-						}
+				if( tile instanceof TileCableBus || tile instanceof TileSkyChest )
+				{
+					return false;
+				}
+
+				final ItemStack[] itemDropCandidates = Platform.getBlockDrops( w, pos );
+				final ItemStack op = new ItemStack( this );
 
-						if( id.removedByPlayer( ids, w, pos, player, false ) )
+				for( final ItemStack ol : itemDropCandidates )
+				{
+					if( Platform.isSameItemType( ol, op ) )
+					{
+						final NBTTagCompound tag = tile.downloadSettings( SettingsFrom.DISMANTLE_ITEM );
+						if( tag != null )
 						{
-							final List<ItemStack> l = Lists.newArrayList( drops );
-							Platform.spawnDrops( w, pos, l );
-							w.setBlockToAir( pos );
+							ol.setTagCompound( tag );
 						}
 					}
+				}
+
+				if( block.removedByPlayer( blockState, w, pos, player, false ) )
+				{
+					final List<ItemStack> itemsToDrop = Lists.newArrayList( itemDropCandidates );
+					Platform.spawnDrops( w, pos, itemsToDrop );
+					w.setBlockToAir( pos );
+				}
+
+				return false;
+			}
+
+			if( heldItem.getItem() instanceof IMemoryCard && !( this instanceof BlockCableBus ) )
+			{
+				final IMemoryCard memoryCard = (IMemoryCard) heldItem.getItem();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method appeng.block.AEBaseTileBlock.onBlockActivated(World, BlockPos, IBlockState, EntityPlayer, EnumHand, ItemStack, EnumFacing, float, float, float) appears to call the same method on the same object redundantly [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)